### PR TITLE
Delete remote intermediates not on local list

### LINKED ISSUE #236: Intermediates collection should be a subset of the ICAs disclosed in CCADB
Every entry in the `intermediates` collection in Remote Settings was, at one time, on a list provided to us by CCADB. Sometimes an intermediate drops off this list, but we keep the corresponding record in Remote Settings. This might be useful for intermediates preloading, but it's currently complicating the signoff logic for CRLite. We should remove entries from `intermediates` which are no longer on the list from CCADB.

### PR DESCRIPTION
**Pull request by jschanck**
Resolves #236 

Note: `remote_expired` constructed on line 555 is a subset of `remote_only`. So there's no need to delete expired certs in a separate loop.

### CODE REVIEW

File: moz_kinto_publisher/main.py [hunk 1]
```diff
@@ -594,18 +594,6 @@ def publish_intermediates(*, args, rw_client, new_filter_run_id=None):
         except KeyError:  # raw_record doesn't have "id"
             log.error(f"Couldn't delete record: {raw_record}")
 
-    # Delete any expired remote records
-    for unique_id in remote_expired:
```
Comment (mozkeeler) [inline moz_kinto_publisher/main.py:598]: So is `remote_expired` used any longer?

**Review by mozkeeler — CHANGES_REQUESTED** [review]
We don't need `remote_expired` any longer, right?

File: moz_kinto_publisher/main.py (hunk 1 above)
Comment (jschanck) [reply moz_kinto_publisher/main.py:598]: Yes, we log the number of expired certs on line 565. I thought about deleting it, but it's nice to have an explanation for some removals.

Comment (mozkeeler) [reply moz_kinto_publisher/main.py:598]: Okay, sounds good.

**Review by mozkeeler — APPROVED** [review]

**Pull request merged**